fence/agents/ironic/fence_ironic.py

```diff
@@ -0,0 +1,138 @@
+#!@PYTHON@ -tt
+
+import atexit
+import logging
+import os
+import re
+import sys
+from pipes import quote
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import fail_usage, is_executable, run_command, run_delay
+
+#BEGIN_VERSION_GENERATION
+RELEASE_VERSION=""
+REDHAT_COPYRIGHT=""
+BUILD_DATE=""
+#END_VERSION_GENERATION
+
+def get_name_or_uuid(options):
+	return options["--uuid"] if "--uuid" in options else options["--plug"]
+
+def get_power_status(_, options):
+    output = ironic_run_command(options, "status")
+    stdout = output[1]
+    match = re.search('power[\\s]*([a-zA-Z]{2,3})', str(stdout))
+    status = match.group(1) if match else None
+    return status
+
+def set_power_status(_, options):
+    ironic_run_command(options, options["--action"])
+    return
+
+def get_devices_list(_, options):
+    nodes = {}
+    output = ironic_run_command(options, "list")
+    stdout = output[1]
+    for line in stdout.splitlines():
+        try:
+            (uuid, name, state) = line.split(',')
+        except ValueError:
+            pass
+        if "UUID" in uuid:
+            continue # skip line header
+        match = re.search('power[\\s]*([a-zA-Z]{2,3})', state)
+        status = match.group(1) if match else None
+        nodes[uuid] = (name, status)
+
+    return nodes
+
+def ironic_run_command(options, action, timeout=None):
```

Comment:
Would it make sense to re-use/generalize
https://github.com/ClusterLabs/fence-agents/blob/master/fence/agents/amt/fence_amt.py#L39 
if yes then we can put it into fencing library.